pycbc/detector.py

```diff
@@ -247,29 +259,27 @@ def set_gmst_reference(self):
 
     def lal(self):
         """ Return lal data type detector instance """
-        if hasattr(self, '_lal'):
-            return self._lal
-        else:
-            import lal
-            d = lal.FrDetector()
-            d.vertexLongitudeRadians = self.longitude
-            d.vertexLatitudeRadians = self.latitude
-            d.vertexElevation = self.info['height']
-            d.xArmAzimuthRadians = self.info['xangle']
-            d.yArmAzimuthRadians = self.info['yangle']
-            d.xArmAltitudeRadians = self.info['yaltitude']
-            d.xArmAltitudeRadians = self.info['xaltitude']
-
-            # This is somewhat abused by lalsimulation at the moment
-            # to determine a filter kernel size. We set this only so that
-            # value gets a similar number of samples as other detectors
-            # it is used for nothing else
-            d.yArmMidpoint = 4000.0
-
-            x = lal.Detector()
-            r = lal.CreateDetector(x, d, lal.LALDETECTORTYPE_IFODIFF)
-            self._lal = r
-            return r
+        import lal
+        d = lal.FrDetector()
+        d.vertexLongitudeRadians = self.longitude
+        d.vertexLatitudeRadians = self.latitude
+        d.vertexElevation = self.info['height']
+        d.xArmAzimuthRadians = self.info['xangle']
+        d.yArmAzimuthRadians = self.info['yangle']
+        d.xArmAltitudeRadians = self.info['xaltitude']
+        d.yArmAltitudeRadians = self.info['yaltitude']
+
+        # This is somewhat abused by lalsimulation at the moment
+        # to determine a filter kernel size. We set this only so that
+        # value gets a similar number of samples as other detectors
+        # it is used for nothing else
+        d.yArmMidpoint = self.info['ylength'] / 2.0
+        d.xArmMidpoint = self.info['xlength'] / 2.0
```

Comment:
Why is this being changed?

Reply:
@spxiwh The lal code doesn't actually use these numbers for much, but it does have them set properly to the right values. This change means that they will be set in the same way as the internal cached detectors in lal in case lal every does use them (for example if someone were to add high-frequency corrections to lalsuite it might matter). Before, I was not bothering to really set them because in the code itself the only thing it used them for was was a guess to a filter length.